Doc updates

GregAC opened the pull request:
I've reworked the first two documentation sections (Overview and User) to be up to date and provide some more info.

More work to be done on the Ibex documentation but this should provide an easier start for those new to the core.

vogelpi commented on doc/01_overview/targets.rst:
```diff
@@ -7,8 +7,6 @@ ASIC Synthesis
 ASIC synthesis is supported for Ibex.
 The whole design is completely synchronous and uses positive-edge triggered flip-flops, except for the register file, which can be implemented either with latches or with flip-flops.
 See :ref:`register-file` for more details.
-The core occupies an area of roughly 24 kGE when using the latch-based register file and implementing the RV32IMC ISA, or 15 kGE when implementing the RV32EC ISA.
-
```
Removing this sounds good to me, especially since we report area numbers in README.md already. I do think we should have some numbers somewhere but I agree that we don't want to keep updating numbers in several places.

vogelpi commented on doc/01_overview/compliance.rst:
```diff
@@ -47,16 +47,17 @@ In addition, the following instruction set extensions are available.
+     - 1.0
+     - always enabled in configurations with PMP see :ref:`PMP Enhancements<pmp-enhancements>`
+
+Ibex currently supports the following features according to the RISC-V Privileged Specification, version 1.12.
```
I don't remember exactly but does Ibex now fully follow version 1.12 of the priv spec?

GregAC replied:
Yes we switched to it a while ago https://github.com/lowRISC/ibex/issues/1733

vogelpi commented on doc/01_overview/verification_overview.rst:
```diff
@@ -0,0 +1,20 @@
+Verification maturity is tracked via :ref:`verification_stages` that are `defined by the OpenTitan project <https://docs.opentitan.org/doc/project/development_stages/#hardware-verification-stages-v>`_.
+Ibex has achieved **V2S** for the `opentitan` configuration, broadly this means verification is almost complete (over 90% code and functional coverage hit with over 90% regression pass rate with test plan and coverage plan fully implemented) but not yet closed.
+
+Nightly regression results are not yet published publicly but we aim to do this soon.
```
"published publicly" -> "released publicly" or just "published"

vogelpi commented on ibex_configs.yaml:
```diff
@@ -2,8 +2,17 @@
 
-# Ibex configurations files, holds the parameter sets that are tested under CI.
-# Each configuration must specify the same set of parameters
+# Ibex named configurations each configuration must specify the same set of
+# parameters
```
Maybe at a comma or line break after `Ibex named configurations`?

marnovandermaas replied:
Nit: missing period

vogelpi commented on doc/02_user/examples.rst:
```diff
@@ -3,16 +3,25 @@
 ========
 
-To make use of Ibex it has to be integrated as described in :ref:`core-integration`.
+There are two examples that demonstrate Ibex usage.
```
I think this commit be merged into the one removing the example from this repo.

GregAC replied:
I see your point, but I think I'll leave them separate just to reduce the amount of rebasing. The commit updating the examples does do more than just alter the reference to the FPGA example too so they'd necessarily go neatly together.

vogelpi commented on doc/02_user/getting_started.rst:
```diff
@@ -3,4 +3,16 @@
 
+There are several paths to follow depending on what you wish to accomplish:
+
+ * See :ref:`examples` for a basic simulation setup running the core in isolation and a simple FPGA system.
```
Obviously, the simple FPGA system lives now the demo repo. But it's probably fine to keep the pointer here.

vogelpi commented on doc/02_user/integration.rst:
```diff
@@ -28,6 +28,63 @@ The vendor ID and implementation ID (``mvendorid`` and ``mimpid``) both read as
+The configuration dependent primitives are:
+ * ``prim_clock_mux2`` - A clock mux, used by the lockstep duplicate core.
+   Required where ``SecureIbex == 1``.
+ * ``prim_flop`` - A multi-bit flop, , used to ensure security critical logic isn't optimized out in synthesis (by applying suitable constraints to prim_flop).
```
I find "multi-bit flop" a bit confusing here because most of these prims are potentially multiple bits wide. I would just write "A flip flop".

vogelpi reviewed (approved):
LGTM, I've just a few minor comments. Thanks @GregAC for woking on this!

marnovandermaas commented on doc/02_user/examples.rst:
```diff
@@ -3,16 +3,25 @@
+
+Simple system is built via FuseSoC.
+Verilator is the primary simulator it is designed for though other simulators are supported (such as VCS).
+Its aim is to make running a binary against Ibex RTL, obtaining a instruction trace, wave trace and any other simulation outputs as simple as possible along with demonstrating basic Ibex integration.
```
Nit: a instruction -> an instruction

marnovandermaas reviewed (commented):
Great work! I just have a couple of comments from an initial read-through

GregAC commented on doc/03_reference/verification.rst:
```diff
@@ -1,3 +1,5 @@
+
 Verification
 ============
 
```
It's not a direct textual copy but yes they do cover similar material. I'd prefer to leave it this way. It's important we have a quick verification overview in 01_overview and equally for anyone who jumps straight to this page to read about verification we can't skip that overview material.

marnovandermaas replied:
Ok, thanks

GregAC commented on doc/02_user/configuration.rst:
```diff
@@ -0,0 +1,43 @@
+
+The current set of supported configurations are:
+
+ * ``small`` - RV32IMC with two stage pipeline and 3 cycle multiplier
+ * ``opentitan`` - The configuration used by the `OpenTitan <www.opentitan.org>`_ project
+ * ``maxperf`` - RV32IMC with three stage pipeline and single cycle multiplier, maximum performance (using stable features) configuration.
+ * ``maxperf-pmp-bmbalanced`` - ``maxperf`` configuration with PMP and the 'balanced' bit-manipulation configuration (:ref:`core-integration` for details).
```
Yes, we should. I think we can just switch to maxperf-pmp-bmbalance, though will need to get some synthesis numbers. We should add the 'micro' configuration too though I don't think that can be a supported configuration as I don't think riscv-dv supports it (I seem to remember there are some toolchain issues with E as well). Still we could do some verilator based smoke test for the micro config, ensure it's not entirely dead ;)

andreaskurth replied:
I've created https://github.com/lowRISC/ibex/issues/1986 to track this

GregAC commented:
Thanks for  the reviews @vogelpi @andreaskurth @marnovandermaas I believe I've now addressed your feedback.

I've also added a new commit that drops the 'EXPERIMENTAL' label from various places in the documentation where it's no longer accurate.

GregAC commented:
CI failures are due to changing config names. It's just private CI, public CI passing (as I can change the configs it runs as part of this PR). I'll switch the config names in private CI once this is merged.

marnovandermaas commented on doc/02_user/configuration.rst:
```diff
@@ -0,0 +1,43 @@
+Supported Configurations
+------------------------
+
+The current set of supported configurations are:
```
What about `maxperf-pmp-bmfull` and `maxperf-pmp-bmfull-icache`?

GregAC replied:
We want to keep the set of supported configs low. We've got testing of the icache in 'opentitan' (as well as the standalone icache testbench). We lack testing for the 'full' bitmanip config, though it is potentially an odd config as it involves a mix of ratified and un-ratified spec (so does 'opentitan' but that it motivated by particular opentitan requirements).

So perhaps we should add a bmfull config at some point, but for now I prefer to just keep things as is. We do need to think about our set of supported bitmanip extensions and what makes sense.

GregAC replied:
I'll keep maxperf-pmp-bmfull in CI, to ensure it doesn't break though will leave it out of supported configurations for now.

marnovandermaas commented on doc/03_reference/pipeline_details.rst:
```diff
@@ -26,7 +26,7 @@ See Multi- and Single-Cycle Instructions below for the details.
 --------------------
 Ibex can be configured to have a third pipeline stage (Writeback) which has major effects on performance and instruction behaviour.
-This feature is *EXPERIMENTAL* and the details of its impact are not yet documented here.
+The details of its impact are not yet documented here.
```
Should we track this in a separate issue?

GregAC replied:
There are various doc updates still to do, including this, I'll create an issue to track everything I can think of.

marnovandermaas reviewed (approved):
LGTM, just two small comments.

andreaskurth reviewed (approved):
LGTM, thanks @GregAC!
